griptape/drivers/vector/mongodb_vector_store_driver.py

```diff
@@ -80,11 +80,56 @@ def load_entries(
             )
 
     def query(
-        self,
-        query: str,
-        count: Optional[int] = None,
-        namespace: Optional[str] = None,
-        include_vectors: bool = False,
-        **kwargs
+            self,
+            query: str,
+            count: Optional[int] = None,
+            namespace: Optional[str] = None,
+            include_vectors: bool = False,
+            offset: Optional[int] = 0,
+            index: Optional[str] = None,
+            **kwargs
     ) -> list[BaseVectorStoreDriver.QueryResult]:
-        raise NotImplementedError("Vector search not supported in MongoDB")
+        collection = self.get_collection()
+
+        # Using the embedding driver to convert the query string into a vector
+        vector = self.embedding_driver.embed_string(query)
+
+        knn_k = count if count else 10
+        pipeline = [
+            {
+                "$search": {
+                    "index": index if index else "default",
+                    "knnBeta": {
+                        "vector": vector,
+                        "path": "vector",
+                        "k": knn_k + offset
+                    }
+                }
+            },
+            {
+                "$project": {
+                    "_id": 1,
+                    "vector": 1,
+                    "namespace": 1,
+                    "meta": 1,
+                    "score": {"$meta": "searchScore"}  # Include the score in the projection
+                }
+            },
+            {"$skip": offset},
+            {"$limit": knn_k}
+        ]
```

Comment:
Just realized, that according to the docs, this doesn't have to be a list, since there is only one element. Can just be a `dict` right?

Reply:
Im using $search and $project here so I do need the list here. It gives me an OperationError without it.
It also gives me - 'A pipeline stage specification object must contain exactly one field.' so using a dictionary would make it multiple items since I'm using $search and $project and that's not allowed.